rust/libnewsboat/src/history.rs

```diff
@@ -0,0 +1,142 @@
+use std::fs::OpenOptions;
+use std::io::{self, BufRead, BufReader, BufWriter, Write};
+use std::path::Path;
+
+pub struct History {
+    idx: usize,
+    lines: Vec<String>,
+}
+
+impl History {
+    pub fn new() -> History {
+        History {
+            idx: 0,
+            lines: Vec::new(),
+        }
+    }
+    pub fn add_line(&mut self, line: String) {
+        if !line.is_empty() {
+            self.lines.insert(0, line)
+        }
+        self.idx = 0;
```

Comment:
In C++, there was a comment here explaining why we need to reset `idx`. Please copy it, because it's really useful to understand what's going on.